app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/DynamicTabAdapter.java

```diff
@@ -398,15 +423,32 @@ public void onClick(View v) {
                         return;
                     }
 
+                    //Hide keypad
+                    hideKeyboard(ctx, v);
+
                     Question question = progressTabStatus.getCurrentQuestion();
                     ReadWriteDB.saveValuesText(question, positiveIntValue);
                     finishOrNext();
                 }
             });
+
         }else{
             editText.setEnabled(false);
             button.setEnabled(false);
         }
+
+        //Add button to listener
+        swipeTouchListener.addClickableView(button);
+
+        //Take focus to this view
+        editText.requestFocus();
+        editText.postDelayed(new Runnable() {
+           @Override
+           public void run() {
+               //Show keypad
+               showKeyboard(ctx, editText);
+           }
+       }, 300); //use 300 to make it run when coming back from lock screen
```

Comment:
This value is the time you wait until the keypad is shown.
A value lower than 300 make it not working properly (for example when
coming back from lock screen.) but it can be adjusted if needed...

2015-09-25 12:57 GMT+02:00 Adrian Quintana <notifications@github.com>:

> In
> app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/DynamicTabAdapter.java
> <https://github.com/EyeSeeTea/pictureapp/pull/70#discussion_r40418882>:
>
> >          }else{
> >              editText.setEnabled(false);
> >              button.setEnabled(false);
> >          }
> > +
> > +        //Add button to listener
> > +        swipeTouchListener.addClickableView(button);
> > +
> > +        //Take focus to this view
> > +        editText.requestFocus();
> > +        editText.postDelayed(new Runnable() {
> > +           @Override
> > +           public void run() {
> > +               //Show keypad
> > +               showKeyboard(ctx, editText);
> > +           }
> > +       }, 300); //use 300 to make it run when coming back from lock screen
>
> @rhardjono <https://github.com/rhardjono> can you explain me a little bit
> more what this 300 means?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EyeSeeTea/pictureapp/pull/70/files#r40418882>.
>